packages/sync-actions/src/product-actions.js

```diff
@@ -401,6 +406,30 @@ function toVariantIdentifier(variant) {
   return id ? { variantId: id } : { sku }
 }
 
+function _buildVariantChangeAssetOrderAction(
+  diffAssets,
+  oldVariant,
+  newVariant
+) {
+  const isAssetOrderChanged = Object.entries(diffAssets).find((entry) =>
+    getIsItemMovedAction(entry[0], entry[1])
+  )
+  if (!isAssetOrderChanged) {
+    return []
+  }
+  const previousAssetIds = oldVariant.assets.map((_) => _.id)
+  const currentAssetIds = newVariant.assets
+    .map((_) => _.id)
+    .filter((_) => _ !== undefined)
```

Comment:
Why we we iterate twice?

```suggestion
  const currentAssetIds = newVariant.assets
    .filter((asset) => asset && asset.id !== undefined)
```